src/devices/src/virtio/block/device.rs

```diff
@@ -329,6 +329,18 @@ impl Block {
     pub fn is_root_device(&self) -> bool {
         self.root_device
     }
+
+    pub fn set_queue(&mut self, idx: usize, q: Queue) {
+        self.queues[idx] = q;
+    }
+
+    pub fn set_rate_limiter(&mut self, rl: RateLimiter) {
+        self.rate_limiter = rl;
+    }
+
+    pub fn rate_limiter(&mut self) -> &RateLimiter {
+        &self.rate_limiter
```

Comment:
Since `self.queues` is `pub(crate)` and these setters and getters are only used in tests, I propose you move their definition to `block::test_utils`